test/unit/connection_test.rb

```diff
@@ -50,5 +51,35 @@
     it 'should return kind of TrainPlugins::Habitat::HTTPGateway' do
       conn.habitat_client.must_be_kind_of TrainPlugins::Habitat::HTTPGateway
     end
+
+    it 'returns a new instance when cache is disabled' do
+      conn.disable_cache(:api_call)
+      client_one = conn.habitat_client
+      client_two = conn.habitat_client
+
+      cache[:api_call].count.must_equal 0
+      client_one.object_id.wont_equal client_two.object_id
+    end
+
+    it 'returns a new instance when cache is enabled and is not hydrated' do
+      cache[:api_call].count.must_equal 0
+      conn.enable_cache(:api_call)
+
+      client_one = conn.habitat_client
+      client_two = conn.habitat_client
+
+      cache[:api_call].count.must_equal 1
+      client_one.object_id.must_equal client_two.object_id
```

Comment:
This test looks a lot like the next test. I'd be OK combining them into one test (removing one or the other) or tweaking this test to remove references to client_two.